### ISSUE TITLE
Keyboard navigation doesn't work properly with an initially empty grid with items added via notifications

### ISSUE THREAD

**Issue by kfranqueiro** [SitePen/dgrid#1259]
This is sort of tangential to #1181, but has definite repercussions for usability.  In the test case below, you cannot tab into the individual item that was added - it still focuses `contentNode`.

```html
<!DOCTYPE html>
<html>
<head>
<link rel="stylesheet" href="../css/dgrid.css">
</head>
<body>
<div id="grid"></div>
<script src="../../dojo/dojo.js" data-dojo-config="async: true"></script>
<script>
require([
	'dojo/_base/declare',
	'dgrid/OnDemandGrid',
	'dgrid/Keyboard',
	'dstore/Memory',
	'dstore/Trackable'
], function (
	declare,
	OnDemandGrid,
	Keyboard,
	Memory,
	Trackable
) {
	var store = new (declare([ Memory, Trackable ]))({ data: [] });
	var grid = new (declare([ OnDemandGrid, Keyboard ]))({
		columns: {
			id: 'ID'
		},
		collection: store
	}, 'grid');

	store.addSync({});
});
</script>
</body>
</html>
```

**Comment by kfranqueiro** [comment]
Pushed a branch/commit with a fix for this.  I'll mull it over and merge next week or the week after (probably won't have much time for dgrid next week).

I ended up resolving this using an `_onNotification` aspect.  I used `aspect` because it fit in better with the existing code, and aspected on `_onNotification` mainly to avoid trouble / edge cases with `removeRow` if I were to try aspecting that directly, given that the main use case here involves stores anyway.

**Comment by kfranqueiro** [comment]
I'm merging my commit since I've had success testing this fix in an app for the past week (in addition to the unit tests passing).
